packages/dds/tree/src/shared-tree/treeCheckout.ts

```diff
@@ -428,32 +342,6 @@ export class TreeCheckout implements ITreeCheckoutFork {
 
 	private readonly views = new Set<TreeView<ImplicitFieldSchema>>();
 
-	public readonly transaction: ITransaction;
-	private readonly transactions = new TransactionStack();
-	/**
-	 * After pushing a starting revision to the transaction stack, this branch might be rebased
-	 * over commits which are children of that starting revision. When the transaction is committed,
-	 * those rebased-over commits should not be included in the transaction's squash commit, even though
-	 * they exist between the starting revision and the final commit within the transaction.
-	 *
-	 * Whenever `rebaseOnto` is called during a transaction, this map is augmented with an entry from the
-	 * original merge-base to the new merge-base.
-	 *
-	 * This state need only be retained for the lifetime of the transaction.
-	 *
-	 * TODO: This strategy might need to be revisited when adding better support for async transactions.
-	 * Since:
-	 *
-	 * 1. Transactionality is guaranteed primarily by squashing at commit time
-	 * 2. Branches may be rebased with an ongoing transaction
-	 *
-	 * a rebase operation might invalidate only a portion of a transaction's commits, thus defeating the
-	 * purpose of transactionality.
-	 *
-	 * AB#6483 and children items track this work.
-	 */
-	private readonly initialTransactionRevToRebasedRev = new Map<RevisionTag, RevisionTag>();
```

Comment:
@yann-achard-MS I removed this because:
1. It isn't used right now. We currently error if we try to rebase during a transaction.
2. It's not clear that we ever will use it (it might be irrelevant after future work).
3. Even if we did use it, I'm not sure I would write it the same way.

I've added a TODO in the relevant new code as a reminder, linked to the work item, and also left a comment on the work item reminding us about this problem, in case it ever matters.